File: libs/eave-stdlib-ts/src/core-api/operations/jira.ts
import { Team } from '../models/team.js';
import { JiraInstallation, QueryJiraInstallationInput, RegisterJiraInstallationInput } from '../models/jira.js';
import { RequestArgsOrigin, makeRequest } from '../../lib/requests.js';
import { sharedConfig } from '../../config.js';

export type RegisterJiraInstallationRequestBody = {
  jira_integration: RegisterJiraInstallationInput;
}
export type RegisterJiraInstallationResponseBody = {
  team?: Team;
  jira_integration: JiraInstallation;
}

export async function registerJiraInstallation({ origin, input }: RequestArgsOrigin & {input: RegisterJiraInstallationRequestBody}): Promise<RegisterJiraInstallationResponseBody> {
  const resp = await makeRequest({
    origin,
    url: `${sharedConfig.eaveApiBase}/integrations/jira/register`,
    input,
  });
  const responseData = <RegisterJiraInstallationResponseBody>(await resp.json());
  return responseData;
}

export type QueryJiraInstallationRequestBody = {
  jira_integration: QueryJiraInstallationInput;
}
export type QueryJiraInstallationResponseBody = {
  team?: Team;
  jira_integration: JiraInstallation;
}

export async function queryJiraInstallation({ origin, input }: RequestArgsOrigin & {input: QueryJiraInstallationRequestBody}): Promise<QueryJiraInstallationResponseBody> {
  const resp = await makeRequest({
    origin,
    url: `${sharedConfig.eaveApiBase}/integrations/jira/query`,
    input,
  });
  const responseData = <QueryJiraInstallationResponseBody>(await resp.json());
  return responseData;
}

File: libs/eave-stdlib-py/src/eave/stdlib/core_api/operations/jira.py
from typing import Optional
import uuid
from eave.stdlib.core_api.models import team
from eave.stdlib.core_api.models.jira import JiraInstallation, RegisterJiraInstallationInput

from . import BaseRequestBody, BaseResponseBody, Endpoint, EndpointConfiguration

from eave.stdlib.eave_origins import EaveOrigin

from ... import requests

class RegisterJiraIntegrationRequest(Endpoint):
    config = EndpointConfiguration(
        path="/integrations/jira/register",
        auth_required=False,
        team_id_required=False,
        signature_required=True,
        origin_required=True,
    )

    class RequestBody(BaseRequestBody):
        jira_integration: RegisterJiraInstallationInput

    class ResponseBody(BaseResponseBody):
        team: Optional[team.Team]
        jira_integration: JiraInstallation

    @classmethod
    async def perform(cls,
        origin: EaveOrigin,
        input: RequestBody,
    ) -> ResponseBody:
        response = await requests.make_request(
            url=cls.config.url,
            origin=origin,
            input=input,
        )

        response_json = await response.json()
        return cls.ResponseBody(**response_json)


# class QueryJiraInstallation(Endpoint):
#     config = EndpointConfiguration(
#         path="/integrations/jira/query",
#         auth_required=False,
#         team_id_required=False,
#         signature_required=True,
#         origin_required=True,
#     )

#     class RequestBody(BaseRequestBody):
#         jira_integration: QueryJiraInstallationInput

#     class ResponseBody(BaseResponseBody):
#         team: Optional[team.Team]
#         jira_integration: Optional[JiraInstallation]

#     @classmethod
#     async def perform(cls,
#         origin: EaveOrigin,
#         input: RequestBody,
#     ) -> ResponseBody:
#         response = await requests.make_request(
#             url=cls.config.url,
#             origin=origin,
#             input=input,
#         )

#         response_json = await response.json()
#         return cls.ResponseBody(**response_json)
